### ISSUE TITLE
Compilation error: 'yield' is not a member of 'std::this_thread'

### ISSUE THREAD

**Issue by joerg-krause** [medoc92/libupnpp#19]
After bumping the version of libupnpp from version 0.15.1 to 0.15.2 our Buildroot buildbot produces the following error for some platforms (sh4a, powerpc):

```
libupnpp/control/discovery.cxx: In constructor 'UPnPClient::UPnPDeviceDirectory::UPnPDeviceDirectory(time_t)':
libupnpp/control/discovery.cxx:338:5: error: 'yield' is not a member of 'std::this_thread'
```

[This](http://autobuild.buildroot.net/results/b6d/b6d7437e1cf8cf0b3726395401ac4ebd9af2833b/build-end.log) is the build log for the sh4a target..
[This](http://autobuild.buildroot.net/results/b6d/b6d7437e1cf8cf0b3726395401ac4ebd9af2833b/libupnpp-0.15.2/config.log) is the config log for the sh4a target.

[This](http://autobuild.buildroot.net/results/0bf/0bf84f61a72a65a75c5fd0a787680f483cfead26/build-end.log) is the build log for the powerpc target..
[This](http://autobuild.buildroot.net/results/0bf/0bf84f61a72a65a75c5fd0a787680f483cfead26/libupnpp-0.15.2/config.log) is the config log for the powerpc target.

Both targets use a GCC 4.7 toolchain, which might not have `std::this_thread::yield()` available.

If you don't care about this, I'm happy to add a dependency on at least GCC 4.8 for libupnpp in Buildroot.

**Comment by medoc92** [comment]
According to the following stackoverflow()  question, you could try to add a def for _GLIBCXX_USE_SCHED_YIELD in the CPPFLAGS.

https://stackoverflow.com/questions/9140019/yield-is-not-a-member-of-stdthis-thread

If you get an undefined  __gthread_yield() at link time, we will know that this is not the right approach, in which case, yes, adding a dependency on gcc 4.8 would be right.

As far as I can see, prior to 4.8, yield is not defined if the above flag is not set. For 4.8 and after, it is defined as an empty function if the flag is not set. (see /usr/include/c++/x.y/thread)

**Comment by joerg-krause** [comment]
`_GLIBCXX_USE_SCHED_YIELD` is an internal macro and should not be used. The problem is, that the toolchains where configured without `--enable-libstdcxx-time`. According to [this](https://gcc.gnu.org/ml/libstdc++/2012-05/msg00085.html), most users and distros do not enable the option. So, `sched_yield()` (`yield()` inlines to `sched_yield()`) cannot be used though it may be available. 

According to [this](https://stackoverflow.com/a/12961816) message, the issue is fixed in GCC 4.9. Therefore, I will propably add a dependency on GCC 4.9 for this package.

**Comment by medoc92** [comment]
In 4.8 std::this_thread::yield() is defined but empty if _GLIBCXX_USE_SCHED_YIELD is not defined (they just moved the #ifdef from outside to inside the function). So I think that 4.8 is ok (yield is defined, but a noop which should not be an issue).
